index.js

```diff
@@ -1,21 +1,131 @@
 'use strict';
 const {app, BrowserWindow} = require('electron');
 
-module.exports = async (url, {type = 'json', ...options} = {}) => {
+let mainWindow;
+let isReady = false;
+
+module.exports = async (url, options = {}) => {
 	await app.whenReady();
 
-	const mainWindow = new BrowserWindow({
-		show: false,
-		sandbox: true
+	if (!mainWindow) {
+		mainWindow = new BrowserWindow({
+			show: false,
+			sandbox: true
+		});
+
+		mainWindow.loadURL('about:blank');
+		await mainWindow.webContents.executeJavaScript(ProxyFetch.toString());
+		await mainWindow.webContents.executeJavaScript('window.fetcher = new ProxyFetch();');
+		isReady = true;
+	}
+
+	while (!isReady) { // eslint-disable-line no-unmodified-loop-condition
+		await delay(30); // eslint-disable-line no-await-in-loop
+	}
```

Comment:
What is this for? I'm sure there's a better way than polling.

Reply:
Check the test called "double fetch without await".
The last `executeJavaScript` must be executed only after the `ProxyFetch` is initialized inside the browser. If you do double fetch without await then `win` variable is already defined and the second fetch will not wait for `ProxyFetch` initialization.

I think this bug is just caused by a slow cpu (my case), without "ready check" with a total of two `executeJavaScript` calls the problem doesn't happen but with three or more then it appears again.

Anyways, as you said, there's a better way than polling and I already implemented it.
Sorry the long explanation, I don't know how to resume it better.